### ISSUE TITLE
request: implement -- (end of options)

### ISSUE THREAD

**Issue by dicktyr** [exiftool/exiftool#38]
please implement `--` end of options
so exiftool can accept filenames beginning with hyphen

```
% exiftool -ver
11.84

% exiftool -file.jpg 
Invalid TAG name: "file.jpg"
No file specified

% exiftool -- -file.jpg 
Unknown option --
```

"The first -- argument that is not an option-argument should be accepted as a delimiter indicating the end of options. Any following arguments should be treated as operands, even if they begin with the '-' character." –[POSIX Specification](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_02)

of course this means giving up on the following feature
(but it's not clear to me how it's useful)

	"Contrary to standard practice, options may appear after 
	source file names on the exiftool command line."

in any case
thanks for sharing your software!

**Comment by boardhead** [comment]
Is it not easy enough to prefix the filename with "./" ?

**Comment by boardhead** [comment]
I should mention that using an end-of-options option like this would disable some important ExifTool features such as the single-line-multiple-command feature (ie. the -execute option)

**Comment by dicktyr** [comment]
> Is it not easy enough to prefix the filename with "./" ? ie.
> 
> `% exiftool ./-file.jpg`

easy enough in manual usage
but less so in automated integration

> I should mention that using an end-of-options option like this would disable some important ExifTool features such as the single-line-multiple-command feature (ie. the -execute option)

I didn't realise that

**Comment by boardhead** [comment]
You've closed this issue, but I'm still thinking about it.  For simple commands, and especially for scripting use, I could see this as a useful feature.

**Comment by dicktyr** [comment]
thank you I agree :}

**Comment by boardhead** [comment]
I'm thinking of adding it for the next release, even though it can't be used with some of the advanced features.

**Comment by boardhead** [comment]
This is implemented in Exiftool 11.87 (just released)
